packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -354,6 +362,124 @@ bool _checkXcodeVersion() {
   return true;
 }
 
+final RegExp _securityFindIdentityDeveloperIdentityExtractionPattern =
+    new RegExp(r'^\s*\d+\).+"(.+Developer.+)"$');
+final RegExp _securityFindIdentityCertificateCnExtractionPattern = new RegExp(r'.*\(([a-zA-Z0-9]+)\)');
+final RegExp _certificateOrganizationalUnitExtractionPattern = new RegExp(r'OU=([a-zA-Z0-9]+)');
+
+/// Given a [BuildableIOSApp], this will try to find valid development code
+/// signing identities in the user's keychain prompting a choice if multiple
+/// are found.
+///
+/// Will return null if none are found, if the user cancels or if the Xcode
+/// project has a development team set in the project's build settings.
+Future<String> getCodeSigningIdentityDevelopmentTeam(BuildableIOSApp iosApp) async{
+  if (iosApp.buildSettings == null)
+    return null;
+
+  // If the user already has it set in the project build settings itself,
+  // continue with that.
+  if (isNotEmpty(iosApp.buildSettings['DEVELOPMENT_TEAM'])) {
+    printStatus('Automatically signing iOS for device deployment using specified development team in Xcode project: ${iosApp.buildSettings['DEVELOPMENT_TEAM']}');
+    return null;
+  }
+
+  if (isNotEmpty(iosApp.buildSettings['PROVISIONING_PROFILE']))
+    return null;
+
+  // If the user's environment is missing the tools needed to find and read
+  // certificates, abandon. Tools should be pre-equipped on macOS.
+  if (!exitsHappy(<String>['which', 'security'])
+      || !exitsHappy(<String>['which', 'openssl']))
+    return null;
+
+  final List<String> validCodeSigningIdentities = runCheckedSync(<String>[
+    'security',
+    'find-identity',
+    '-p',
+    'codesigning',
+    '-v',
+  ])
+      .split('\n')
+      .map<String>((String outputLine) {
+        return _securityFindIdentityDeveloperIdentityExtractionPattern.firstMatch(outputLine)?.group(1);
+      })
+      .where((String identityCN) => isNotEmpty(identityCN))
+      .toSet() // Unique.
+      .toList();
+
+  final String signingIdentity = _chooseSigningIdentity(validCodeSigningIdentities);
+
+  // If none are chosen.
+  if (signingIdentity == null)
+    return null;
+
+  printStatus('Signing iOS app for device deployment using developer identity: "$signingIdentity"');
+
+  final String signingCertificateId =
+      _securityFindIdentityCertificateCnExtractionPattern.firstMatch(signingIdentity)?.group(1);
+
+  // If `security`'s output format changes, we'd have to update this
+  if (signingCertificateId == null)
+    return null;
+
+  final String signingCertificate = runCheckedSync(<String>[
+    'security',
+    'find-certificate',
+    '-c',
+    signingCertificateId,
+    '-p',
+  ]);
+
+  final Process opensslProcess = await runCommand(<String>['openssl', 'x509', '-subject']);
```

Comment:
Commented on https://github.com/dart-lang/sdk/issues/29590 -- the following may work (adapted for `flutter_tools` APIs):

```
Process findCertificate = await runCommand(<String>['security', ...]);
Process openssl = await runCommand(<String>['openssl', ...]);
findCertificate.stdout.pipe(openssl.stdin);
String opensslOutput = await UTF8.decodeStream(openssl.stdout);

if (await openssl.exitCode != 0) {
  printError('...');
  return null;
}
```